File: AspNetMvcUnleashed/0672329980_source/Chapter21/VB/UnleashedBlog.Tests/Controllers/ApplicationControllerTests.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting

<TestClass()> _
Public Class ApplicationControllerTests
    Private _controllerFactory As ControllerFactory
    Private _applicationController As ApplicationController

    <TestInitialize()> _
    Public Sub Initialize()
        _controllerFactory = New ControllerFactory()
        _applicationController = _controllerFactory.GetApplicationController()
    End Sub

    ''' <summary>
    ''' Verifies that ArchiveInfo (used by the ArchiveMenu) gets into view data
    ''' </summary>
    <TestMethod()> _
    Public Sub AddsArchiveInfoToViewData()
        ' Assert
        Assert.IsInstanceOfType(_applicationController.ViewData("ArchiveInfo"), GetType(IList(Of ArchiveInfo)))
    End Sub
End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter21/CS/UnleashedBlog/UnleashedBlog.Tests/Controllers/ApplicationControllerTests.cs
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using UnleashedBlog.Controllers;
using UnleashedBlog.Models;
using UnleashedBlog.Tests.Factories;

namespace UnleashedBlog.Tests.Controllers
{

    [TestClass]
    public class ApplicationControllerTests
    {
        private ControllerFactory _controllerFactory;
        private ApplicationController _applicationController; 

        [TestInitialize]
        public void Initialize()
        {
            _controllerFactory = new ControllerFactory();
            _applicationController = _controllerFactory.GetApplicationController();
        }

        /// <summary>
        /// Verifies that ArchiveInfo (used by the ArchiveMenu) gets into view data
        /// </summary>
        [TestMethod]
        public void AddsArchiveInfoToViewData()
        {
            // Assert
            Assert.IsInstanceOfType(_applicationController.ViewData["ArchiveInfo"], typeof(IList<ArchiveInfo>));
        }
    }
}
